crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -835,6 +836,20 @@ impl Stage for SystemStage {
 
             // Reevaluate run criteria.
             has_work = false;
+            match stage_should_run {
+                ShouldRun::No => (),
+                ShouldRun::Yes => stage_should_run = ShouldRun::No,
+                ShouldRun::YesAndCheckAgain | ShouldRun::NoAndCheckAgain => {
+                    stage_should_run = self.stage_run_criteria.should_run(world);
+                    match stage_should_run {
+                        ShouldRun::Yes | ShouldRun::YesAndCheckAgain => has_work = true,
+                        ShouldRun::No | ShouldRun::NoAndCheckAgain => (),
+                    }
+                }
+            }
+            if let ShouldRun::YesAndCheckAgain = stage_should_run {
```

Comment:
doesn't this double-run stage_run_criteria if line 843 returns YesAndCheckAgain

Reply:
Yes... yes it does. The perils of rushed single-commit bugfixes. 6e7a51bd7956e9f7d71f261a9c00c9e4a8190731 should take care of that.